src/commands/app/deploy.js

```diff
@@ -83,7 +84,18 @@ class Deploy extends BuildCommand {
           throw error
         }
       }
-      // 2. deploy actions and web assets for each extension
+
+      // 2. Bail if workspace is production and application status is PUBLISHED, honor force-deploy
+      if (aioConfig.project.workspace.name === 'Production' && flags.publish && !flags['force-deploy']) {
```

Comment:
`flags.publish` in the condition here should be removed. The whole publish wording is misleading. see comment below.